Automattic/studio
# Studio: Snapshot counter not updating in settings when a site is deleted

Fixes https://github.com/Automattic/dotcom-forge/issues/8132

## Proposed Changes

This diff fixes a bug when the snapshot counter in user settings is not updated when the user deletes a site from settings tab.


NOTES:
1. The problem was created due to circular dependencies between the different hooks and `useSiteDetails`.
2. The issue is now fixed, by centralizing all snapshot logic in one place, and use it in useSiteDetails, where necessary
3. It can simplified further, and moved into different hooks to offload the main `use-snapshots` provider. However, as we have moved to another project, I leave it for a different ticket.


## Testing Instructions
1. Create a site
5. Add a snapshot
6. Open user settings and ensure the counter has been increased
7. Navigate to settings
8. Press "Delete site" button at the bottom of the page
9. Ensure that after visiting settings, the counter has been decreased.



## Pre-merge Checklist
